app/src/main/java/ru/netology/custom_view/ui/StatsView.kt

```diff
@@ -110,26 +133,35 @@ class StatsView @JvmOverloads constructor(
     }
 
     override fun onDraw(canvas: Canvas) {
-        if (data.isEmpty()) {
+        if (data.isEmpty() || progress == 0F) {
             return
         }
-
-        var startFrom = -90F
+        var startFrom = INITIAL_ANGLE
+        val maxAngle = 360 * progress
```

Comment:
Вот тут не совсем правда. Вы забыли добавить `INITIAL_ANGLE`. У вас получилось, что `maxAngle` не учитывает тот факт, что система отсчёта начинается не с `0`, а с `-90`. Подумайте ещё тоже над этим. Вроде как тогда дальнейшие вычисления должны пойти проще

Reply:
Тут я не совсем понимаю, что вы имеете ввиду и как это изменить.
Я руководствуюсь следующей логикой:
        // Проверим не вышли ли за пределы с отрисовкой следующего фрагмента
        if (startFrom - INITIAL_ANGLE + angle > maxAngle) {
            drawData(
                index = index,
                canvas = canvas,
                startFrom = startFrom,
                // Корректировка
                sweepAngle = maxAngle - startFrom + INITIAL_ANGLE
            )
            // Хватит)
            return
        }
        drawData(
            index = index,
            canvas = canvas,
            startFrom = startFrom,
            sweepAngle = angle
        )
        startFrom += angle
    }

Reply:
Обратите внимание, что у Вас везде, где встречается `maxAngle` присутствует так же `... + INITIAL_ANGLE`. На самом деле даже в условии `startFrom - INITIAL_ANGLE + angle > maxAngle` присутствует эта запись, потому что она равносильна `startFrom + angle > maxAngle + INITIAL_ANGLE`. 
Итого, достаточно в самом начале при расчёте `maxAngle` прибавить к нему `INITIAL_ANGLE` и дальше забыть про эту константу.
```suggestion
        val maxAngle = 360 * progress + INITIAL_ANGLE
```

Reply:
Исправлено